src/lib/openjp2/openjpeg.c

```diff
@@ -33,7 +33,30 @@
 
 #ifdef _WIN32
 #include <windows.h>
-#endif /* _WIN32 */
+#include <fcntl.h>
+
+#else /* _WIN32 */
+
+#include <errno.h>
+
+#include <stdarg.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+
+#ifdef HAVE_UNISTD_H
+# include <unistd.h>
+#endif
+
+#ifdef HAVE_FCNTL_H
```

Comment:
Neither HAVE_FCNTL_H nor HAVE_IO_H will work without a corresponding CMake test.
There is an example in thirdparty/libtiff/tif_config.h.cmake.in.